src/dialect/redshift.rs

```diff
@@ -41,14 +41,48 @@ impl Dialect for RedshiftSqlDialect {
     /// treating them as json path. If there is identifier then we assume
     /// there is no json path.
     fn is_proper_identifier_inside_quotes(&self, mut chars: Peekable<Chars<'_>>) -> bool {
```

Comment:
Wondering why redshift is special with this, Im thinking since there are other dialects that support jsonpath (which implements partiql in the parser) it shouldn't be. Would it not be enough to remove this `is_proper_identifier_inside_quotes` method entirely and the special `[` character from `is_delimited_identifier_start`?
From what I can tell in [their docs](https://docs.aws.amazon.com/redshift/latest/dg/r_names.html), `[` isn't an identifier delimiter so it seems that having the dialect `is_delimited_identifier_start` flag otherwise is incorrect?

Reply:
It is a reasonable question!

I verified what happened if `is_proper_identifier_inside_quotes` is removed and `[` is removed from RedShift dialect - a couple tests failed `brackets_over_db_schema_table_name_with_whites_paces` and `test_square_brackets_over_db_schema_table_name` + my new one. Both are related to an old fashion way of using `[` in RedShift.

I verified that these queries work in RedShift and we can find it in query history or in end-user usages:
1. `SELECT 1 AS ["1"] FROM foo`
2. `SELECT [a] FROM [ schema ].[foo]`

Perhaps, needs to add a special case for `[` in RedShift dialect instead of improving identifying a possible identifier inside quotes.

Reply:
@iffyio I have updated the PR based on your suggestion. Now, a common quoted identifier is processed without additional conditions, and RedShift uses the same set of starting characters for quoted identifiers as PostgreSQL.

Additionally, I added a specific match to handle the RedShift-specific case for [. All tests passed without requiring any changes.